Code/Framework/AzManipulatorTestFramework/Include/AzManipulatorTestFramework/ActionDispatcher.h

```diff
@@ -79,89 +79,104 @@ namespace AzManipulatorTestFramework
         virtual void SetSelectedEntityImpl(AZ::EntityId entity) = 0;
         virtual void SetSelectedEntitiesImpl(const AzToolsFramework::EntityIdList& entities) = 0;
         virtual void EnterComponentModeImpl(const AZ::Uuid& uuid) = 0;
-        template <typename... Args>
-        void Log(const char* format, const Args&... args);
+        void Log(const char* format, ... );
         bool m_logging = false;
+
     private:
         const char* KeyboardModifierString(const AzToolsFramework::ViewportInteraction::KeyboardModifier& keyModifier);
     };
 
-    template <typename DerivedDispatcherT>
-    template <typename... Args>
-    void ActionDispatcher<DerivedDispatcherT>::Log(const char* format, const Args&... args)
+    template<typename DerivedDispatcherT>
+    void ActionDispatcher<DerivedDispatcherT>::Log(const char* format, ...)
     {
+        va_list args;
+        va_start(args, format);
+
         if (m_logging)
         {
-            AZ_Printf("ActionDispatcher", format, args...);
+            vprintf(format, args);
+            printf("%s", "\n");
         }
+
+        va_end(args);
     }
 
-    template <typename DerivedDispatcherT>
-    template <typename... Args>
-    DerivedDispatcherT* ActionDispatcher<DerivedDispatcherT>::Trace(const char* format, const Args&... args)
+    template<typename DerivedDispatcherT>
+    DerivedDispatcherT* ActionDispatcher<DerivedDispatcherT>::Trace(const char* format, ...)
     {
-        Log(format, args...);
+        va_list args;
```

Comment:
What was wrong with the version which was changed? It seems to cut out some code duplication.

Reply:
It switched to use the c-style vaargs instead of a variadic template - this it to work with `printf` instead of `AZ_Printf`. Annoyingly you can't do a pass-through call with `...` which is why there's the small amount of repetition (which is next to each other so I decided it's acceptable). The alternative is to write a recursive variadic template to mimic `printf`, but it seemed like a bunch of extra complexity and effort for not much gain.

Reply:
OK cool, thanks for the explanation.

Reply:
Alternative suggestion: Write you own local print function that does support `...` and that one will call `print` internally, that way you don't have to change the function templates, it's just a matter of calling `AZ_TestPrintf` (suggestion) instead of `AZ_Printf`. And in the future, when `AZ_Printf` works with tests it'll be a much simpler change.

Reply:
`AZ_Printf` was explicitly disabled for tests so I don't see it coming back anytime soon, what I was trying to explain is the syntax for what you've described isn't possible...

```c++
    template<typename DerivedDispatcherT>
    void ActionDispatcher<DerivedDispatcherT>::Log(const char* format, ...)
    {
        va_list args;
        va_start(args, format);

        if (m_logging)
        {
            vprintf(format, args);
            printf("%s", "\n");
        }

        va_end(args);
    }

    template<typename DerivedDispatcherT>
    DerivedDispatcherT* ActionDispatcher<DerivedDispatcherT>::Trace(const char* format, ...)
    {
        Log(format, ...); // not supported
        return static_cast<DerivedDispatcherT*>(this);
    }
```

You can't do this as far as I'm aware, it's just not supported, which is why we have this small amount of duplication (and don't use variadic templates anymore, which should actually help with compile times in this case)